opt4j-optimizers/src/main/java/org/opt4j/optimizers/ea/NonDominatedFronts.java

```diff
@@ -0,0 +1,178 @@
+	 * @param dominatedIndividualsMap
+	 *            map mapping an individual on the collection of individuals that
+	 *            it dominates
+	 * @param dominatingIndividualNumber
+	 *            an array where the number of dominating individuals is stored
+	 *            for each individual
+	 * @param individual2IndexMap
+	 *            a map storing the indices of the individuals used to access the
+	 *            dominatingIndividualNumber
+	 * @return the list of individuals forming the next non-dominated front
+	 */
+	protected List<Individual> getNextFront(List<Individual> currentFront,
+			Map<Individual, List<Individual>> dominatedIndividualsMap, int[] dominatingIndividualNumber,
+			Map<Individual, Integer> individual2IndexMap) {
+		List<Individual> nextFront = new ArrayList<Individual>();
+		for (Individual dominant : currentFront) {
+			for (Individual dominated : dominatedIndividualsMap.get(dominant)) {
+				dominatingIndividualNumber[individual2IndexMap.get(dominated)]--;
+				if (dominatingIndividualNumber[individual2IndexMap.get(dominated)] == 0) {
+					nextFront.add(dominated);
+				}
+			}
+		}
+		return nextFront;
+	}
+
+	/**
+	 * Compare all possible individual pairs. For each individual, store 1) the
+	 * number of individuals it is dominated by and 2) the set of individuals it
+	 * dominates.
+	 * 
+	 * @param individualList
+	 *            a list of the individuals
+	 * @param dominatedIndividualsMap
+	 *            A map that is filled during the execution of the method. Each
+	 *            individual is mapped onto the set of individuals that are
+	 *            dominated by this individual.
+	 * @param dominatingIndividualNumber
+	 *            An integer array (initialized with zeros) that is filled during
+	 *            the execution of this method. Each individual is associated with
+	 *            an entry of this array. The integer therein is the number of
+	 *            individuals this individual is dominated by.
+	 * @param individual2IndexMap
+	 *            a map mapping each individual onto its index in the
+	 *            dominatingIndividualNumber - array
+	 */
+	protected void determineDomination(List<Individual> individualList,
+			Map<Individual, List<Individual>> dominatedIndividualsMap, int[] dominatingIndividualNumber,
+			Map<Individual, Integer> individual2IndexMap) {
+		// compare each individual with each other individual
+		for (int i = 0; i < individualList.size(); i++) {
+			for (int j = i + 1; j < individualList.size(); j++) {
+				Individual p = individualList.get(i);
+				Individual q = individualList.get(j);
+				Objectives po = p.getObjectives();
+				Objectives qo = q.getObjectives();
+				if (po.dominates(qo)) {
+					dominatedIndividualsMap.get(p).add(q);
+					dominatingIndividualNumber[individual2IndexMap.get(q)]++;
+				} else if (qo.dominates(po)) {
+					dominatedIndividualsMap.get(q).add(p);
+					dominatingIndividualNumber[individual2IndexMap.get(p)]++;
+				}
```

Comment:
What happens if neither po dominates qo nor qo dominates po?
Perhaps, add an else case just with the docu why we can ignore this case?